module/zfs/dbuf.c

```diff
@@ -879,9 +892,10 @@ dbuf_kstat_update(kstat_t *ksp, int rw)
 void
 dbuf_init(void)
 {
-	uint64_t hsize = 1ULL << 16;
+	uint64_t hmsize, hsize = 1ULL << 16;
 	dbuf_hash_table_t *h = &dbuf_hash_table;
-	int i;
+
+	memset(h, 0, sizeof (dbuf_hash_table_t));
```

Comment:
Is this needed?  Aren't all global symbols are zeroed automatically?  I think they are.

Reply:
They are automatically zeroed, but it's still needed by `ztest` because `ztest_freeze()` does the following.  

```
kernel_init()
kernel_fini()
kernel_init()
kernel_fini()
```

which leaves the global initialized after the `kernel_fini()`.  We can can either zero the struct in `kernel_init()` before it's used, or afterwards in `kernel_fini()`.  I opted for `kernel_init()` which seemed cleanest, but I don't feel strongly about it.

Reply:
Ah. OK then.

Reply:
We could do `#ifndef _KERNEL` around it with a comment that it is for ztest, although it is a very minor thing.

Reply:
I went ahead and tweaked the patch to instead set `hash_table` and `hash_mutexes` to `NULL` before their respective while loops.  This seems little nicer to me and achieves the same thing.

```
       h->hash_table = NULL;
       while (h->hash_table == NULL) {
       }
```